packages/sanity/src/structure/panes/document/timeline/events/EventsTimeline.tsx

```diff
@@ -0,0 +1,263 @@
+import {Box, Flex, Spinner, Stack, Text} from '@sanity/ui'
+import {motion, type Variants} from 'framer-motion'
+import {useCallback, useEffect, useMemo, useState} from 'react'
+import {
+  CommandList,
+  type CommandListRenderItemCallback,
+  type DocumentGroupEvent,
+  getDocumentVariantType,
+  isCreateDocumentVersionEvent,
+  isEditDocumentVersionEvent,
+  LoadingBlock,
+  useTranslation,
+} from 'sanity'
+
+import {ExpandableTimelineItemButton} from '../expandableTimelineItemButton'
+import {ListWrapper, Root} from '../timeline.styled'
+import {EventTimelineItem} from './EventTimelineItem'
+import {PublishedEventMenu} from './PublishedEventMenu'
+
+interface TimelineProps {
+  events: DocumentGroupEvent[]
+  hasMoreEvents: boolean | null
+  selectedEventId?: string
+  onLoadMore: () => void
+  onSelect: (event: DocumentGroupEvent) => void
+  onExpand: (event: DocumentGroupEvent) => Promise<void>
```

Comment:
It's async because we show a loading state while the parent is expanding if the transactions are not available yet. 
It would go like this:
- user visits a draft or version document
- user sees a `Published` event (we don't have the transactions for that published event)
- User clicks on the **expand** icon
  - We fetch the transactions that made up to that event and we show a loading state.
  - Transactions load, we show the edit events.

https://github.com/user-attachments/assets/9e3f4a52-82ff-4025-8e3c-cabc715554a0

I think naming could be improved to indicate the async nature of the action, will do that